utility/with_size_reader.go

```diff
@@ -0,0 +1,40 @@
+package utility
+
+import (
+	"io"
+	"sync/atomic"
+)
+
+func NewWithSizeReader(underlying io.Reader, readSize *int64) *WithSizeReader {
```

Comment:
In most cases, modifying value by pointer is not a go way.
It would be better to create a separate method like `GetBytesRead() int` to return counter.

But, as far as I see, this code is used in tests and passing `*int` is handy there.
May be just move this code to `_test.go` files ?